src/Elasticsearch.Net/Extensions/NameValueCollectionExtensions.cs

```diff
@@ -11,35 +11,38 @@ internal static void CopyKeyValues(this NameValueCollection source, NameValueCol
 		{
 			foreach (var key in source.AllKeys)
 			{
-				if (dest[key] != null) throw new Exception(string.Format("Attempted to add duplicate key '{0}'", key));
+				if (dest[key] != null) throw new Exception($"Attempted to add duplicate key '{key}'");
 
 				dest.Add(key, source[key]);
 			}
 		}
+
 		internal static string ToQueryString(this NameValueCollection self, string prefix = "?")
 		{
 			if (self == null)
 				return null;
 
-			if (self.AllKeys.Length == 0) return string.Empty;
-			
-			return prefix + string.Join("&", self.AllKeys.Select(key => $"{Encode(key)}={Encode(self[key])}"));
+			return self.AllKeys.Length == 0
+				? string.Empty
+				: $"{prefix}{string.Join("&", self.AllKeys.Select(key => $"{Encode(key)}={Encode(self[key])}"))}";
 		}
 
+		//TODO: It's not placeholder function ;)
 		private static string Encode(string s) => s;
 		//private static string Encode(string s) => s == null ? null : Uri.EscapeDataString(s);
 
-		internal static NameValueCollection ToNameValueCollection(this IDictionary<string, object> dict, IFormatProvider provider)
+		internal static NameValueCollection ToNameValueCollection(this IDictionary<string, object> originalDictionary,
+			IFormatProvider provider)
 		{
-			if (dict == null || dict.Count < 0)
+			if (originalDictionary == null || originalDictionary.Count < 0)

```

Comment:
In case originalDictionary=null statement if(originalDictionary?.Count < 0) will equal false.

After that in line in `originalDictionary.Wher(` would be NullRefereanceException